src/views/product-details.vue

```diff
@@ -264,10 +268,37 @@ export default defineComponent({
       isJobPending: 'job/isJobPending',
       jobTotal: 'job/getTotal',
       userProfile: 'user/getUserProfile',
-      selectedBrand: 'user/getSelectedBrand'
-    })
+      selectedBrand: 'user/getSelectedBrand',
+    }),
+    filteredProducts () {
+      const filteredProducts = JSON.parse(JSON.stringify(this.current));
+      if(this.filter.size.length || this.filter.color.length){
+        filteredProducts.list.items = [];
+        filteredProducts.list.items = this.current.list.items.map((item: any)=>{
+          const product = this.getProduct(item.groupValue);
+          const hasSize = this.filter.size.some((sizeFeature: any) => {
+            return product.productFeatures.includes("Size/" + sizeFeature)
+          })
+          const hasColor = this.filter.color.some((colorFeature: any) => {
+            return product.productFeatures.includes("Color/" + colorFeature)
+          })
+          if (hasSize || hasColor) return item;
+          else return null
+        }).filter((product: any) => product);
+      }
+      return filteredProducts;
+    }
   },
   methods: {
+    filterVarients (featureValue: any, type: any) {
+      if(this.filter[type].includes(featureValue)){
```

Comment:
Fix spacing issues with if and else both

Reply:
Fixed.